### PR TITLE
[FLINK-9942][rest] Guard handlers against null fields

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR fixes prevents some NPEs that could arise if fields in the request are set to null or are omitted.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobSubmitRequestBody.java [hunk 1]
```diff
@@ -34,7 +35,7 @@
  */
 public final class JobSubmitRequestBody implements RequestBody {
 
-	private static final String FIELD_NAME_JOB_GRAPH = "jobGraphFileName";
+	public static final String FIELD_NAME_JOB_GRAPH = "jobGraphFileName";
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobSubmitRequestBody.java]: That would _work_ but probably wouldn't return a useful error message.

Generally speaking I'd prefer if by default all fields are required (which was also the case at some point), waiting for @GJL response on why that was changed.

[the remark this answers is not in the document]
Comment (dawidwys) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobSubmitRequestBody.java]: Ignore my comments. I've checked it and the `required` property is ignored by the `ObjectMapper`

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobSubmitRequestBody.java [hunk 2]
```diff
@@ -53,8 +54,16 @@ public JobSubmitRequestBody(
 			@JsonProperty(FIELD_NAME_JOB_JARS) Collection<String> jarFileNames,
```
Comment (dawidwys) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobSubmitRequestBody.java]: Maybe annotate `jobGraphFileName` with `@Nonnull`?

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**